src/blueprint/definitions.rs

```diff
@@ -522,35 +522,43 @@ pub fn to_definitions<'a>() -> TryFold<'a, ConfigModule, Vec<Definition>, String
     TryFold::<ConfigModule, Vec<Definition>, String>::new(|config_module, _| {
         let output_types = config_module.output_types();
         let input_types = config_module.input_types();
-        Valid::from_iter(config_module.types.iter(), |(name, type_)| {
-            let dbl_usage = input_types.contains(name) && output_types.contains(name);
-            if let Some(variants) = &type_.variants {
-                if !variants.is_empty() {
-                    to_enum_type_definition(name, type_, variants).trace(name)
+        Valid::from_iter(
+            config_module
+                .types
+                .iter()
+                .collect::<BTreeMap<_, _>>()
+                .iter(),
```

Comment:
This was necessary as `HashMap` is more optimized for reads but is unordered, while `BTreeMap` has ordering.

Instead of manually ordering the `HashMap` to be used, I'm collecting into the ordered map.